services/configuration/src/api.rs

```diff
@@ -199,35 +190,29 @@ impl PublicConfigApi {
                     config_hash
                 ))
             })
-            .filter(|config| {
-                PublicConfigApi::filter_cfg_predicate(
-                    config,
-                    previous_cfg_hash_filter,
-                    actual_from_filter,
-                )
-            })
-            .map(|config| self.get_config_with_proofs(config))
+            .filter(|config| filter.test(config))
+            .map(|config| self.config_with_proofs(config))
             .collect();
         committed_configs
```

Comment:
Looks it's possible to remove `let committed_configs` variable, because it is immediately returned?

Reply:
Regarding this and other places that have silenced `let_and_return` lint: The code *won't* compile without an additional variable, because the borrow checker thinks that the index is still borrowed after `collect()` (but not in the case the returned value is bound to a variable). I can't explain why this happens, and I would like to get rid of it.

Reply:
Fascinating! I also can't explain what's going on here, but `#![feature(nll)]` fixes this, so I think this might be just a bug/limitation of the current borrow checker. 

However, it looks like the iterators here don't actually need lifetimes to start with? Both `MemoryDB` and `RocksDB` don't actually make use of the lifetime.